src/countsHelper.cpp

```diff
@@ -0,0 +1,31 @@
+#include <Rcpp.h>
+#include <unordered_map>
+#include <vector>
+using namespace Rcpp;
+
+//' @title
+//' Count number of occurences
+//'
+//' @param features a Matrix with doubles.
+//'
+//' @export
+// [[Rcpp::export]]
+NumericMatrix  countsToDhist(NumericMatrix features)
```

Comment:
Does this really take a matrix rather than a vector? It looks like we assume dimension = 1 except for rows.

Reply:
Short answer, it does assume that, happy to swap it to being a vector.